src/fluree/db/query/fql.cljc

```diff
@@ -621,14 +621,18 @@
                (if inVector? [res'] res'))
 
              (let [{:keys [headers tuples]} (if aggregates (analytical/add-aggregate-cols res aggregates) res)
-                   offset' (if (or selectDistinct? groupBy) 0 offset)
-                   limit'  (if selectDistinct? nil group-limit)
-                   tuples' (if orderBy
-                             (order-offset-and-limit-results tuples headers orderBy offset' limit')
-                             tuples)
-                   res'    (->> (dissoc opts :limit :offset :orderBy :groupBy)
-                                (format-filter-tuples db tuples' select-spec headers vars)
-                                <?)]
+                   offset'  (if groupBy 0 offset)
```

Comment:
question: Do you want to check for either groupBy or orderBy to set the offset' to 0?  It looks like it could be applied twice in the conditions below.

orderBy (order-offset-and-limit-results tuples' headers orderBy offset' group-limit)
(or offset' group-limit) (->> tuples' (drop offset') (take group-limit))

Reply:
Not sure I follow... I don't see where that would happen. This is a cond, not a cond->>... so the conditions are exclusive.